### PR TITLE
added npmingnore, upgraded to react 16, configured es6 lint

### PR DESCRIPTION
**Pull request by okezieokpara**
Add a .npmignore file, upgraded the example project to react 16 and configured linting with eslint

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class ExampleComponent extends Component {
 
     return (
       <div className={styles.test}>
-        Example Component: {text}
+        Example Common Net: {text}
```
Comment (transitive-bullshit) [inline src/index.js:22]: I'm guessing this is an accidental change?

**Comment by transitive-bullshit** [comment]
Hey @okezieokpara, I really appreciate the work.

One general piece of advice when making PRs would be to not change the original author's style, as those types of PRs usually never get adopted.

With that being said, there are definitely some good bits in here I'd love to merge, so let's break the changes down:
- definitely agree with upgrading the example to latest version of react and react-scripts
- `.npmignore` is redundant because `package.json` specifies `files`. npmignore was previously removed in a previous [PR](https://github.com/transitive-bullshit/react-modern-library-boilerplate/pull/10) for this reason.
- having both `eslintConfig` and `.eslintrc` is redundant. You only need to specify one, and I see no reason to not use the current eslintrc.
- adding semicolons everywhere and changing the expected style is perfectly acceptable in your fork, but this boilerplate will continue to enforce JS standard style.

**Comment by transitive-bullshit** [comment]
I've gone ahead and updated the example app's react and react-scripts to latest.

**Comment by okezieokpara** [comment]
Thanks, Travis, I may not have double checked a few things before committing.
Aside, I am quite new to Rollup and can't configure it to import from ".jsx" files.

**Comment by transitive-bullshit** [comment]
@okezieokpara it's all good -- I really appreciate your interest!

Regarding `.jsx` files, I'm not sure why that wouldn't work.  Maybe try importing from the full filename like this:

```js
import test from 'component.jsx'
```

**Pull request closed without merging**